llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp

```diff
@@ -808,6 +812,29 @@ void WaitcntBrackets::setScoreByOperand(const MachineInstr *MI,
   setScoreByInterval(Interval, CntTy, Score);
 }
 
+// Return true if the subtarget is one that enables Point Sample Acceleration
+// and the MachineInstr passed in is one to which it might be applied (the
+// hardware makes this decision based on several factors, but we can't determine
+// this at compile time, so we have to assume it will be applied if the
```

Comment:
We don't know whether the optimization will be applied or not, i.e. whether the instruction will be ordered like a load or ordered like a sample.
```suggestion
// this at compile time, so we have to assume it might be applied if the
```

Reply:
Done